File: Codeforces/Round 484/A.cpp
/*input
8
10010101
*/

// author : ShivamKD

#include <bits/stdc++.h>

using namespace std;

#define FIO ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
#define ll long long
const ll MOD = 1e9 + 7;

int main(){
	FIO
	int n;
	cin>>n;
	string s;
	cin>>s;
	int first = s[0] - '0';
	int count = first;
	for(int i = 1;i<n;i++){

		if(s[i]=='1'){
			count++;
			if(s[i-1]=='1'){
				cout<<"No";
				return 0;
			}
		}
	}
	if(count== (n/2 + n%2)){
		cout<<"Yes";
	}
	else
		cout<<"No";
	return 0;
}

File: Codeforces/Round 484/A.py
'''input
4
0100
'''

def ii():
	return int(input())
def ai():
	return map(int,input().split())

n = ii()
s = input()
if n== 1 :
	if s == "1":
		print("Yes")
	else:
		print("No")
	exit(0)
#print(s[n-2:n])
#print(c)
#print(n // 2 + n % 2)
if "11" in s or "000" in s:
	print("No")
else :
	if s[0 : 2] =="00" or s[n-2:n] == "00":
		print("No")
	else:
		print("Yes")

